test/integration/asyncprofiler_test.go

```diff
@@ -0,0 +1,138 @@
+	cmd := exec.Command("java", "-Xms1G", "-Xmx1G", "-XX:+AlwaysPreTouch", "-jar", javaProgramPath)
+	fmt.Fprintf(os.Stdout, "Executing command: %v\n", cmd)
+	err = cmd.Start()
+	require.NoError(t, err, "Failed to start Java program")
+
+	time.Sleep(50 * time.Second) // We need to wait for the Java program to start
+	// Get the Java process PID
+	javaPID := cmd.Process.Pid
+	fmt.Fprintf(os.Stdout, "Java PID: %d\n", javaPID)
+
+	// Initialize AsyncProfiler with required parameters
+	profiler := asyncprofiler.NewAsyncProfiler(
+		filepath.Join(cwd, "testdata", "jattach"),
+		filepath.Join(cwd, "testdata", "libasyncProfiler.so"),
+		javaPID,
+	)
+
+	// Run fdtransfer
+	err = asyncprofiler.RunFdtransfer(filepath.Join(cwd, "testdata", "fdtransfer"), javaPID)
+	require.NoError(t, err, "Failed to run fdtransfer")
+
+	time.Sleep(10 * time.Second) // Wait for fdtransfer to finish loading
+
+	jfrOutputFile := filepath.Join(cwd, "prof.jfr")
+	// Set action to start
+	err = profiler.SetAction("start", asyncprofiler.ProfilerOptions{
+		"event": "cpu",
+		"file":  jfrOutputFile,
+	})
+	require.NoError(t, err, "Failed to set action")
+
+	// Construct the start command
+	ctx, cancel := context.WithTimeout(context.Background(), 120*time.Second)
+	defer cancel()
+	profilerCmd, err := profiler.BuildCommand(ctx)
+	require.NoError(t, err, "Failed to build command")
+
+	// Execute the start command
+	var outbuf, errbuf bytes.Buffer
+	profilerCmd.Stdout = &outbuf
+	profilerCmd.Stderr = &errbuf
+	fmt.Fprintf(os.Stdout, "Executing command: %v\n", profilerCmd)
+	if err = profilerCmd.Run(); err != nil {
+		err = fmt.Errorf("Failed to execute profiler: %w: %s: %s", err, outbuf.String(), errbuf.String())
+	}
+	require.NoError(t, err, "Failed to run start command")
+
+	// Sleep for the specified duration
+	time.Sleep(120 * time.Second)
+
+	// Set action to stop
+	err = profiler.SetAction("stop")
+	require.NoError(t, err, "Failed to set action")
+
+	// Construct the stop command
+	ctx, cancel = context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+	profilerCmd, err = profiler.BuildCommand(ctx)
+	require.NoError(t, err, "Failed to build command")
+
+	// Execute the stop command
+	fmt.Fprintf(os.Stdout, "Executing command: %v\n", profilerCmd)
+	err = profilerCmd.Run()
+	require.NoError(t, err, "Failed to run stop command")
+
+	// Check the exit code
+	require.Equal(t, 0, profilerCmd.ProcessState.ExitCode(), "Non-zero exit code")
+
+	// Check if the output file "prof.jfr" exists
+	_, err = os.Stat(jfrOutputFile)
+	require.NoError(t, err, "Failed to find output file")
+
+	jfrFile, err := os.Open(jfrOutputFile)
+	require.NoError(t, err, "Failed to open JFR file")
+	defer jfrFile.Close()
+
+	pprofProfile, err := convert.JfrToPprof(jfrFile)
+	require.NoError(t, err, "Failed to convert JFR to pprof")
+	require.NotNil(t, pprofProfile, "Pprof profile is nil")
+
+	// Save the converted pprof profile to a file
+	pprofFile, err := os.Create("prof.pprof")
```

Comment:
I don't think we need to do this, we can stop at the point where JRF to pprof conversion doesn't fail. However, we should check that the profile created is not empty (as in the `.Sample` is non 0 length).